src/D2L.CodeStyle.Analyzers/Immutability/ImmutabilityContext.Factory.cs

```diff
@@ -77,24 +78,23 @@ internal partial class ImmutabilityContext {
 			("System.IO.Abstractions.FileSystem", default)
 		);
 
+
+		internal static readonly ImmutableArray<(string TypeName, string MethodName, string AssemblyName)> KnownImmutableReturningMethods = ImmutableArray.Create(
+			( "System.Array", "Empty", default(string) ),
+			( "System.Linq.Enumerable", "Empty", default(string) )
+		);
+		
+
 		internal static ImmutabilityContext Create( Compilation compilation, AnnotationsContext annotationsContext ) {

```

Comment:
This `Create` function should be getting the `IMethodSymbol`s in `KnownImmutableReturningMethods` (`KnownImmutableReturningMethods` should be `private`) and then passing them in to `ImmutabilityContext`, so that getting the symbols is only done once.

Reply:
Explained differently, we're not looking to share ImmutabilityContext's loading code, we're instead looking to expand the scope of what ImmutabilityContext knows about.